content/en/blog/2023/http-conventions-declared-stable.md

```diff
@@ -0,0 +1,249 @@
+| `net.peer.port` &rarr; `server.port` | Now captured even when same as default port for scheme |
+{.td-initial .table .table-responsive .ot-table-first-row-60}
+<!-- prettier-ignore-end -->
+
+References:
+
+- [HTTP Client span attributes v1.20.0](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.20.0/specification/trace/semantic_conventions/http.md#http-client)
+- [HTTP Client span attributes v1.23.0 (stable)](https://github.com/open-telemetry/semantic-conventions/blob/v1.23.0/docs/http/http-spans.md#http-client)
+
+### HTTP server span attributes
+
+<!-- prettier-ignore-start -->
+| Change | Comments |
+| --- | --- |
+| `http.route` | No change |
+| `http.target` &rarr; `url.path` and `url.query` | Split into two separate attributes |
+| `http.scheme` &rarr; `url.scheme` | Now factors in [X-Forwarded-Proto][], [Forwarded#proto][] headers |
+| `http.client_ip` &rarr; `client.address` | If `http.client_ip` was unknown (i.e., no [X-Forwarded-For][], [Forwarded#for][] headers), then `net.sock.peer.addr` &rarr; `client.address`; now must be provided to sampler |
+| `net.host.name` &rarr; `server.address` | Now based only on [Host][Host header], [:authority][HTTP/2 authority], [X-Forwarded-Host][], [Forwarded#host][] headers |
+| `net.host.port` &rarr; `server.port` | Now based only on [Host][Host header], [:authority][HTTP/2 authority], [X-Forwarded-Host][X-Forwarded-Host], [Forwarded#host][] headers |
+{.td-initial .table .table-responsive .ot-table-first-row-50}
+<!-- prettier-ignore-end -->
+
+References:
+
+- [HTTP Server span attributes v1.20.0](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.20.0/specification/trace/semantic_conventions/http.md#http-server)
+- [HTTP Server span attributes v1.23.0 (stable)](https://github.com/open-telemetry/semantic-conventions/blob/v1.23.0/docs/http/http-spans.md#http-server)
+
+### HTTP client and server span names
+
+The `{http.method}` portion of span names is replace by `HTTP` when
+`{http.method}` is `_OTHER`.
+
+See note if [migrating from `<= v1.17.0`](#migrating-from--v1170).
+
+References:
+
+- [HTTP client and server span names v1.20.0](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.20.0/specification/trace/semantic_conventions/http.md#name)
+- [HTTP client and server span names v1.23.0 (stable)](https://github.com/open-telemetry/semantic-conventions/blob/v1.23.0/docs/http/http-spans.md#name)
+
+### HTTP client duration metric
+
+<!-- prettier-ignore-start -->
+| Change | Comments |
+| --- | --- |
+| Name: `http.client.duration` &rarr; `http.client.request.duration` | |
+| Unit: `ms` &rarr; `s` | Unit changed from milliseconds to seconds |
+| Description | Updated from "_measuring the duration of inbound HTTP requests_" &rarr; the "_duration of HTTP server requests_" |
+| Histogram buckets | Boundaries updated to reflect change from milliseconds to seconds, and zero bucket boundary removed |
```

Comment:
do you think this helps?

```suggestion
| **Name**: `http.client.duration` &rarr; `http.client.request.duration` | |
| **Unit**: `ms` &rarr; `s` | Unit changed from milliseconds to seconds |
| **Description** | Updated from "_measuring the duration of inbound HTTP requests_" &rarr; the "_duration of HTTP server requests_" |
| **Histogram buckets** | Boundaries updated to reflect change from milliseconds to seconds, and zero bucket boundary removed |
```

Reply:
Sure that's a good idea.
I _almost_ kept them separate: e.g., leaving those properties in a list and only putting the attributes rows in the table. Would you prefer that?

Reply:
> I _almost_ kept them separate: e.g., leaving those properties in a list and only putting the attributes rows in the table. Would you prefer that?

this sounds good 👍

Reply:
I'm on it!